### PR TITLE
llvm: Drop bundled libc++abi as it may cause run time issues with system libc++abi.

### LINKED ISSUE #47149: llvm: libcxx in LLVM formula sometimes causes problems
When LLVM formula is installed with libcxx, `libc++.dylib` and `libc++abi.dylib` will be generated inside `/usr/local/opt/llvm/lib`. If a program wants to link against LLVM, the compile command line may look like:
~~~
clang++ -o main -std=c++11 -L/usr/local/opt/llvm/lib -I ... main.cpp
~~~
In this case, `clang` will use `libc++` in `/usr/local/opt/llvm/lib` instead of OS X's `libc++`, and give compile errors:
~~~
Undefined symbols for architecture x86_64:
  "std::terminate()", referenced from:
      ___clang_call_terminate in main-1455b4.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
~~~
This is because `libc++.dylib` in `/usr/local/opt/llvm/lib` does not correctly export symbols of `libc++abi`.
Adding `-lc++abi` can make the program succeed to compile. But when executing it, following error is given:
~~~
dyld: Library not loaded: @rpath/libc++abi.1.dylib
  Referenced from: /usr/local/opt/llvm/lib/libc++.1.dylib
  Reason: image not found
fish: './main' terminated by signal SIGTRAP (Trace or breakpoint trap)
~~~
This error is caused by `rpath` is not set when compiling the program, i.e. `rpath` should be set to `/usr/local/opt/llvm/lib`.
Therefore, `libc++.dylib` and `libc++abi.dylib` in `/usr/local/opt/llvm/lib` can be really annoyed if the program wants to link LLVM. In most case, we do not intend to link LLVM's `libc++` instead of system's. Also, even if we do not care which `libc++` to link, linking LLVM's `libc++` needs more compiler flags.

My suggestion is to remove `libc++` from LLVM formula. Maybe we can create an individual formula for it.

### PR DESCRIPTION
**Pull request by manphiz**
Fixes #47149.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
I think I'd rather we just drop this option for reasons in the previous thread.

**Comment by manphiz** [comment]
@mikemcquaid OK, I'll update the PR.

**Comment by manphiz** [comment]
@mikemcquaid Updated.

File: Library/Formula/llvm.rb [hunk 1]
```diff
@@ -236,10 +230,20 @@ def install
...
+    s = <<-EOS.undent
       LLVM executables are installed in #{opt_bin}.
       Extra tools are installed in #{opt_share}/llvm.
     EOS
+
+    if build.with? "libcxx"
+      s += <<-EOS.undent
+        To use the bundled libc++, please adjust your LDFLAGS as following:
+
+        LDFLAGS="-L#{opt_lib} -lc++abi"
```
Comment (mikemcquaid) [inline Library/Formula/llvm.rb]: This need updated/removed?

Comment (manphiz) [inline Library/Formula/llvm.rb]: Well as @zhipeng-jia noted in the issue, the built libc++ doesn't add indirect symbols for libc++abi, so `-lc++abi` is still required, even with system libc++abi.

Comment (mikemcquaid) [inline Library/Formula/llvm.rb]: Cool, just wanted to check as it wasn't obvious, thanks!

**Comment by manphiz** [comment]
@mikemcquaid Seems all travis-ci tests timed out for some reason. Should they be retried?

**Comment by UniqMartin** [comment]
It's simply a long build. You can safely ignore Travis CI failures due to timeouts if the default build is fine.

**Comment by manphiz** [comment]
@mikemcquaid ping :)

File: Library/Formula/llvm.rb [hunk 2]
```diff
@@ -236,10 +230,20 @@ def install
...
   def caveats
-    <<-EOS.undent
+    s = <<-EOS.undent
       LLVM executables are installed in #{opt_bin}.
       Extra tools are installed in #{opt_share}/llvm.
     EOS
+
+    if build.with? "libcxx"
+      s += <<-EOS.undent
+        To use the bundled libc++, please adjust your LDFLAGS as following:
```
Comment (mikemcquaid) [inline Library/Formula/llvm.rb]: `To use the bundled libc++ please add the following LDFLAGS:`

File: Library/Formula/llvm.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/llvm.rb]: Ident these by two characters and remove the empty line above.

**Comment by mikemcquaid** [comment]
@manphiz Sorry, final nits and good to go!

**Comment by mikemcquaid** [comment]
@manphiz A few final tweaks and we're good to go. Can you also adjust the commit message to the `llvm: ...` format and make it not longer than 50 characters on the first line? Thanks!

**Comment by manphiz** [comment]
@mikemcquaid Thanks for the comments. Adjusted and let's wait for the checks to complete. Sorry for AFK for the past few days.

**Comment by mikemcquaid** [comment]
Thanks again @manphiz!

**Comment by manphiz** [comment]
@mikemcquaid My pleasure, Mike!

**Pull request closed without merging**